src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1152,6 +1152,95 @@ private string ReplaceHelper(int oldValueLength, string newValue, ReadOnlySpan<i
+
+        /// <summary>
+        /// Replaces all newline sequences in the current string with <paramref name="replacementText"/>.
+        /// </summary>
+        /// <returns>
+        /// A string whose contents match the current string, but with all newline sequences replaced
+        /// with <paramref name="replacementText"/>.
+        /// </returns>
+        /// <remarks>
+        /// This method searches for all newline sequences within the string and canonicalizes them to the
+        /// newline sequence provided by <paramref name="replacementText"/>. If <paramref name="replacementText"/>
+        /// is <see cref="string.Empty"/>, all newline sequences within the string will be removed.
+        ///
+        /// It is not recommended that protocol parsers utilize this API. Protocol specifications often
+        /// mandate specific newline sequences. For example, HTTP/1.1 (RFC 8615) mandates that the request
+        /// line, status line, and headers lines end with CRLF. Since this API operates over a wide range
+        /// of newline sequences, a protocol parser utilizing this API could exhibit behaviors unintended
+        /// by the protocol's authors.
+        ///
+        /// The list of recognized newline sequences is CR (U+000D), LF (U+000A), CRLF (U+000D U+000A),
+        /// NEL (U+0085), LS (U+2028), FF (U+000C), and PS (U+2029). This list is given by the Unicode
+        /// Standard, Sec. 5.8, Recommendation R4 and Table 5-2.
+        ///
+        /// This method is guaranteed O(n * r) complexity, where <em>n</em> is the length of the input string,
+        /// and where <em>r</em> is the length of <paramref name="replacementText"/>.
+        /// </remarks>
+        public string ReplaceLineEndings(string replacementText)
+        {
+            if (replacementText is null)
+            {
+                throw new ArgumentNullException(nameof(replacementText));
+            }
+
+            // Early-exit: do we need to do anything at all?
+            // If not, return this string as-is.
+            // See comment in GetIndexOfFirstNewLineChar for why we don't use IndexOfAny.
+
+            int idxOfFirstNewlineChar = NewLineUtility.GetIndexOfFirstNewLineChar(this, out int charsToConsume);
+            if (idxOfFirstNewlineChar < 0)
+            {
+                return this;
+            }
+
+            // Don't bother memcpying the first segment into the builder.
+            // Use the builder only for the subsequent segments, then use
+            // the final string.Concat call to prepend the first segment once
+            // we know what the final string length will be.
+
+            ReadOnlySpan<char> firstSegment = this.AsSpan(0, idxOfFirstNewlineChar);
+
+            ValueStringBuilder builder = new ValueStringBuilder(stackalloc char[256]);
+            foreach (ReadOnlySpan<char> nextSegment in this.AsSpan(idxOfFirstNewlineChar + charsToConsume).EnumerateLines())
```

Comment:
>this.AsSpan(idxOfFirstNewlineChar + charsToConsume) [](http://example.com/codeflow?start=55&length=51)

interesting, I didn't know that `string.AsSpan(string.Length)` doesn't fail and it looks it return empty span. I am seeing you already having a test covering such case e..g "<CR>" case.